using-gh-actions-and-okteto-cloud-to-preview-pr-changes.md

```diff
@@ -0,0 +1,185 @@
+      uses: okteto/login@master
+      with:
+        token: ${{ secrets.OKTETO_TOKEN }}
+
+    - name: Create namespace
+      uses: okteto/create-namespace@master
+      with:
+        namespace: pr-${{ github.event.number }}-namespace
+   
+    - name: Deploy application
+      uses: okteto/deploy-stack@master
+      with:
+        build: "true"
+        namespace: pr-${{ github.event.number }}-namespace
+
+    - name: add comment to PR
+      uses: unsplash/comment-on-pr@master
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        msg: "Preview environment available at https://fastapi-pr-${{ github.event.number }}-namespace.cloud.okteto.net"  
+```
+
+In the code block above, we defined the job `preview` which runs on the `ubuntu-latest` environment. The subtasks to be executed under this job is defined under the `steps` heading. Let's go over these subtasks.
+
+Replace `namespace` in the code block above with your Okteto default namespace.
+
+### Subtasks
+
+In our `preview` job, we have five tasks to be executed. The first task checks out the repository using the `actions/checkout@master` actions, so that the repository is accessible by the entire workflow. 
+
+The second task logs in into your Okteto account as a prerequisite to creating a new namespace where the preview environment will be deployed. This task makes use of a token `secrets.OKTETO_TOKEN` which is your Okteto token. In the next section, you'll learn how to add your okteto token to your repository.
+
+The third task creates a new namespace using the `okteto/create-namespace@master` action and the fourth task, uses the action `okteto/deploy-stack@master` to deploy the application using the okteto stack file. Once again, we have witnessed the adaptability of Okteto stacks.
+
+The last task adds a comment on the pull request discussion indicating a link to the deployed application. It makes use of the `unsplash/comment-on-pr@master` action.
+
+### Adding Your Okteto Token To GitHub
+
+Before adding your Okteto token to GitHub, retrieve your OKteto token from your dashboard settings:
+
+![Okteto dashboard - Settings](https://res.cloudinary.com/laisi/image/upload/v1615047658/pbkydburxszrz8hjgmgg.png)
+
+The workflow file needs your [Okteto account](https://cloud.okteto.com) token to function properly. Follow the steps below to add your okteto token to your GitHub repository:
+
+1. In your repository, navigate to the **Settings** tab and click on the **Secrets** link on the menu by the left:
+
+![Secrets](https://res.cloudinary.com/laisi/image/upload/v1615045117/cskayu5fymd9pfkcnvvn.png)
+
+2. Click on **New Repository Secret**
+
+![Add A New Secret](https://res.cloudinary.com/laisi/image/upload/v1615045246/vvavcscqrww9dnxm6pyh.png)
+
+3. Create a new secret `OKTETO_TOKEN` and set the value to the token you copied from your dashboard.
+
+![Create New Secret](https://res.cloudinary.com/laisi/image/upload/v1615047981/pqfyim20zfw8gzoycwma.png)
+
+
```

Comment:
can we link to the official githhub instructions for using secrets at the end?

Reply:
Sure

Reply:
Done